lib/js_routes.rb

```diff
@@ -216,13 +216,19 @@ def build_js(route, parent_route)
   end
 
   def route_js_arguments(route, parent_spec)
+    parts = route.parts
     required_parts = route.required_parts.clone
-    optional_parts = route.parts - required_parts
-    default_parts = route.defaults.select do |part, _|
+    optional_parts = parts - required_parts
+    default_options = route.defaults.select do |part, _|
       FILTERED_DEFAULT_PARTS.exclude?(part) && URL_OPTIONS.include?(part) || required_parts.include?(part)
     end
     [
-      required_parts, optional_parts, serialize(route.path.spec, parent_spec), default_parts
+      parts,
+      required_parts,
+      optional_parts,
+      default_options,
+      Rails.version < '4.0',
+      serialize(route.path.spec, parent_spec)
```

Comment:
Definition of an Object from [ECMAScript Third Edition (pdf)](http://www.ecma-international.org/publications/files/ECMA-ST-ARCH/ECMA-262,%203rd%20edition,%20December%201999.pdf):

> 4.3.3 Object

> An object is a member of the type Object. It is **an unordered collection of properties** each of which contains a primitive value, object, or function. A function stored in a property of an object is called a method.

So the order of properties in JSON serialization could differ from the order of properties in deserialized object when iterating it using `for`. Order is very crucial as I said.